manage_arkime/commands/add_vpc.py

```diff
@@ -7,19 +7,43 @@
 from manage_arkime.cdk_client import CdkClient
 import manage_arkime.constants as constants
 import manage_arkime.cdk_context as context
+from manage_arkime.vni_provider import SsmVniProvider, VniAlreadyUsed, VniOutsideRange, VniPoolExhausted
 
 logger = logging.getLogger(__name__)
 
-def cmd_add_vpc(profile: str, region: str, cluster_name: str, vpc_id: str, vni: int):
+def cmd_add_vpc(profile: str, region: str, cluster_name: str, vpc_id: str, user_vni: int):
     logger.debug(f"Invoking add-vpc with profile '{profile}' and region '{region}'")
 
     aws_provider = AwsClientProvider(aws_profile=profile, aws_region=region)
+    vni_provider = SsmVniProvider(cluster_name, aws_provider)
 
-    # Confirm the VNI is valid
-    if (vni <= constants.VNI_MIN) or (constants.VNI_MAX < vni):
-        logger.error(f"VNI {vni} is outside the acceptable range of {constants.VNI_MIN} to {constants.VNI_MAX} (inclusive)")
-        logger.warning("Aborting...")
-        return
+    # If the user didn't supply a VNI, try to find one
+    if not user_vni:
+        try:
+            next_vni = vni_provider.get_next_vni()
+        except VniPoolExhausted:
+            logger.error(f"There are no remaining VNIs in the range {constants.VNI_MIN} to {constants.VNI_MAX} to assign for this cluster")
+            logger.warning("Aborting...")
+            return
+
+    # Confirm the user-supplied VNI is available
+    else:
+        next_vni = user_vni
+
+        try:
+            if not vni_provider.is_vni_available(next_vni):
```

Comment:
So is the idea to just leave the errors for now incase we change our mind?

Reply:
Yeah, that's the idea.  I'm trying to keep this layer of the code agnostic of both the assignment strategy and the specific data store.  Similarly, I've positioned the VniProvider code to be able to separate the strategy and data store.  If we ever decide to change the data store from SSM, it shouldn't affect much in the VniProvider code; if we decide to change the strategy OR data store, it shouldn't affect much in this code.